makeStage2.sh

```diff
@@ -40,5 +41,100 @@ cd $usepop/pop/x/src/
+# couldn't find a clear guarantee that env generates a sorted list.)
+#
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g' | \
+    sort
+}
+
+
+### nox ########################################################################
+
+# Newpop - see https://raw.githubusercontent.com/GetPoplog/Base/main/pop/help/newpop
+# Rebuilds $popsys: re-links basepop11, rebuild saved images and generate scripts.
+# -nox specifies basepop11 should not be linked against X-windows.
+$usepop/pop/src/newpop -link -nox -norsv
+
+echo_env "$usepop" > "${BUILD_HOME}/environments/nox-base"
+echo_env "$usepop/pop/.." > "${BUILD_HOME}/environments/nox-base-cmp"
+( cd "${BUILD_HOME}/environments" && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-nox!g' < nox-base | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-nox/!g' > nox-new )
+
+mkdir -p "$usepop"/pop/pop-nox
+mkdir -p "$usepop"/pop/lib/psv-nox
+( cd "$usepop"/pop/pop; tar cf - . ) | ( cd "$usepop"/pop/pop-nox; tar xf - )
+( cd "$usepop"/pop/lib/psv; tar cf - . ) | ( cd "$usepop"/pop/lib/psv-nox; tar xf - )
+
+### motif ######################################################################
+
+# Newpop - see https://raw.githubusercontent.com/GetPoplog/Base/main/pop/help/newpop
+# Rebuilds $popsys: re-links basepop11, rebuild saved images and generate scripts.
+# -x=xm specifies basepop11 should be linked against the Motif-toolkit.
 $usepop/pop/src/newpop -link -x=-xm -norsv
 
+echo_env "$usepop" > "${BUILD_HOME}/environments/xm-base"
+echo_env "$usepop/pop/.." > "${BUILD_HOME}/environments/xm-base-cmp"
+( cd "${BUILD_HOME}/environments" && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xm!g' < xm-base | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xm/!g' > xm-new )
+
+mkdir -p "$usepop"/pop/pop-xm
+mkdir -p "$usepop"/pop/lib/psv-xm
+( cd "$usepop"/pop/pop; tar cf - . ) | ( cd "$usepop"/pop/pop-xm; tar xf - )
+( cd "$usepop"/pop/lib/psv; tar cf - . ) | ( cd "$usepop"/pop/lib/psv-xm; tar xf - )
+
+
+### Xt #########################################################################
+
+# Newpop - see https://raw.githubusercontent.com/GetPoplog/Base/main/pop/help/newpop
+# Rebuilds $popsys: re-links basepop11, rebuild saved images and generate scripts.
+# -x=xt specifies basepop11 should be linked against the X-toolkit.
+$usepop/pop/src/newpop -link -x=-xt -norsv
+
+echo_env "$usepop" > "${BUILD_HOME}/environments/xt-base"
+echo_env "$usepop/pop/.." > "${BUILD_HOME}/environments/xt-base-cmp"
+( cd "${BUILD_HOME}/environments" && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xt!g' < xt-base | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xt/!g' > xt-new )
+
+# Rename rather than copy.
+mv "$usepop"/pop/pop "$usepop"/pop/pop-xt
```

Comment:
Bit easier in bash: `mv "$usepop/pop"/pop{,-xt}` and more clear you're renaming.

Reply:
An idiom I haven't used before. Quite neat.